app/javascript/controllers/tabs_controller.js

```diff
@@ -0,0 +1,41 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="tabs"
+export default class extends Controller {
+  static targets = ["content", "tab"];
+
+  connect() {
+    this.loadContent(this.currentUrl());
+  }
+
+  switchTab(event) {
+    event.preventDefault();
+    const url = event.currentTarget.href;
+    this.loadContent(url);
+    this.updateActiveTab(event.currentTarget);
+  }
+
+  loadContent(url) {
+    fetch(url, {
+      headers: {
+        'X-Requested-With': 'XMLHttpRequest'
+      }
+    })
+    .then(response => response.text())
+    .then(html => {
+      const parser = new DOMParser();
+      const doc = parser.parseFromString(html, 'text/html');
+      const newContent = doc.getElementById('notes-or-transcript').innerHTML;
```

Comment:
I’m a little worried about using the beatified transcript. I think it’ll make it impossible to have a feature where you can tap on a line in the transcript and have it take you to that part of the video. but I think for now this is good.